weave-js/src/components/PagePanelComponents/Home/Browse3/compare/ComparePageObjectsLoaded.tsx

```diff
@@ -0,0 +1,399 @@
+            [RESOLVED_REF_KEY]: r,
+          };
+        }
+      }
+      refValues[r] = val;
+    }
+    let resolved = objects;
+    let dirty = true;
+    const mapper = (context: TraverseContext) => {
+      if (
+        isWeaveRef(context.value) &&
+        refValues[context.value] != null &&
+        // Don't expand _ref keys
+        context.path.tail() !== RESOLVED_REF_KEY
+      ) {
+        dirty = true;
+        return refValues[context.value];
+      }
+      return _.clone(context.value);
+    };
+    while (dirty) {
+      dirty = false;
+      resolved = resolved.map(o => mapObject(o, mapper));
+    }
+    setResolvedData(resolved);
+  }, [objects, refs, refsData.loading, refsData.result]);
+
+  const merged = mergeObjects(objectIds, resolvedData);
+
+  // Hide rows that are uninteresting for diff
+  const uninterestingPaths =
+    objectType === 'call' ? UNINTERESTING_PATHS_CALL : UNINTERESTING_PATHS_OBJ;
+  const filtered = merged.filter(
+    row => !uninterestingPaths.includes(row.path.toString())
+  );
+  const diffed = computeDiff(objectIds, filtered, false);
+
+  // Determine if objects are all the same type. If so, use that
+  // term instead of the generic "object" in title
+  let title = 'Compare calls';
+  if (objectType === 'object') {
+    const baseClasses = objects.map(v => v.baseObjectClass);
+    const allSameType =
+      baseClasses[0] && baseClasses.every(c => c === baseClasses[0]);
+    const baseType = allSameType
+      ? baseClasses[0]?.toLocaleLowerCase()
+      : 'object';
+    title = `Compare ${baseType}s`;
+  }
+
+  const hasModes = objectIds.length === 2 || selected;
+  const checkedMode = hasModes ? mode : 'parallel';
+
+  const cartItems = objects.map(v => {
+    if (objectType === 'call') {
+      return {
+        key: 'call',
+        value: v.id,
+        label: v.id.slice(-4),
+      };
+    }
+    return {
+      key: 'obj',
+      value: `${v.objectId}:v${v.versionIndex}`,
+    };
+  });
+
+  const onSetMode = (newMode: Mode) => {
+    querySetString(history, 'mode', newMode);
+  };
+
+  const setOnlyChanged = () => {
+    queryToggleBoolean(history, 'changed', false);
+  };
+
+  const onSetBaseline = (value: boolean) => {
+    const {search} = history.location;
+    const params = new URLSearchParams(search);
+    if (value) {
+      params.set('baseline', '1');
+    } else {
+      params.delete('baseline');
+    }
+    history.replace({
+      search: params.toString(),
+    });
+  };
+
+  return (
+    <SimplePageLayout
+      title={title}
+      hideTabsIfSingle
+      tabs={[
+        {
+          label: '',
+          content: (
+            <TailwindContents>
+              <div className="flex items-center gap-4 px-16 pt-12">
+                {isSequential && (
+                  <Button
+                    size="small"
+                    disabled={first.version === 0}
+                    icon="back"
+                    variant="ghost"
+                    onClick={onPrev}
```

Comment:
as a user, I am not sure what this does (especially when there are only 2 items?) unclear why these arrows are here, but not in the calls one (without reading the code)